pkg/machine/define/vmfile.go

```diff
@@ -48,6 +48,18 @@ func (m *VMFile) Read() ([]byte, error) {
 	return os.ReadFile(m.GetPath())
 }
 
+// Read the first n bytes of a given file and return in []bytes
+func (m *VMFile) ReadMagicNumber(n int) ([]byte, error) {
+	f, err := os.Open(m.GetPath())
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	b := make([]byte, n)
+	_, err = f.Read(b)
+	return b, err
```

Comment:
```suggestion
	b := make([]byte, n)
	n, err = f.Read(b)
	return b[:n], err
```
this should return `b[:n], err` because in theory we might read less than n.